tests/includes/wait-for.sh

```diff
@@ -331,3 +331,51 @@ wait_for_storage() {
 		sleep "${SHORT_TIMEOUT}"
 	fi
 }
+
+# wait_for_aws_ingress_cidrs_for_port_range blocks until the expected CIDRs
+# are present in the AWS security group rules for the specified port range.
+wait_for_aws_ingress_cidrs_for_port_range() {
+	local from_port to_port exp_cidrs cidr_type
+
+	from_port=${1}
+	to_port=${2}
+	exp_cidrs=${3}
+	cidr_type=${4}
+
+	ipV6Suffix=""
+	if [ "$cidr_type" = "ipv6" ]; then
+		ipV6Suffix="v6"
+	fi
+
+	# shellcheck disable=SC2086
+  secgrp_list=$(aws ec2 describe-security-groups --filters Name=ip-permission.from-port,Values=${from_port} Name=ip-permission.to-port,Values=${to_port})
+  # print the security group rules
+  echo "sg = ${secgrp_list}"
+  # shellcheck disable=SC2086
+  got_cidrs=$(echo ${secgrp_list} | jq -r ".SecurityGroups[0].IpPermissions | .[] | select(.FromPort == ${from_port} and .ToPort == ${to_port}) | .Ip${ipV6Suffix}Ranges | .[] | .CidrIp${ipV6Suffix}" | sort | paste -sd, -)
+
+  echo "expected CIDRs: ${exp_cidrs}"
+	attempt=0
+	# shellcheck disable=SC2046,SC2143
+	until [ "$attempt" -gt "7" ] && [ "$got_cidrs" -eq "$exp_cidrs" ]; do
+		echo "[+] (attempt ${attempt}) polling security group rules"
+		# shellcheck disable=SC2086
+		secgrp_list=$(aws ec2 describe-security-groups --filters Name=ip-permission.from-port,Values=${from_port} Name=ip-permission.to-port,Values=${to_port})
+		# print the security group rules
+		echo "sg = ${secgrp_list}"
+		# shellcheck disable=SC2086
+		got_cidrs=$(echo ${secgrp_list} | jq -r ".SecurityGroups[0].IpPermissions | .[] | select(.FromPort == ${from_port} and .ToPort == ${to_port}) | .Ip${ipV6Suffix}Ranges | .[] | .CidrIp${ipV6Suffix}" | sort | paste -sd, -)
+		sleep "${SHORT_TIMEOUT}"
+		attempt=$((attempt + 1))
+	done
+
+	# shellcheck disable=SC2086
+	secgrp_list=$(aws ec2 describe-security-groups --filters Name=ip-permission.from-port,Values=${from_port} Name=ip-permission.to-port,Values=${to_port})
+	# shellcheck disable=SC2086
+	got_cidrs=$(echo ${secgrp_list} | jq -r ".SecurityGroups[0].IpPermissions | .[] | select(.FromPort == ${from_port} and .ToPort == ${to_port}) | .Ip${ipV6Suffix}Ranges | .[] | .CidrIp${ipV6Suffix}" | sort | paste -sd, -)
+	if [ "$got_cidrs" != "$exp_cidrs" ]; then
+		# shellcheck disable=SC2046
+		echo $(red "expected generated EC2 ${cidr_type} ingress CIDRs for range [${from_port}, ${to_port}] to be:\n${exp_cidrs}\nGOT:\n${got_cidrs}")
+		exit 1
+	fi
```

Comment:
All these lines are going into the while loop, so you need to replace them with a correct exit from the loop and an error thrown otherwise.